### PR TITLE
GH-9635: Add GetUsersByUsernames to plugin API

### PR DESCRIPTION
**Pull request by czertbytes**
#### Summary
add GetUsersByUsernames method to plugin API.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9635

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: app/plugin_api.go [hunk 1]
```diff
@@ -160,6 +160,10 @@ func (api *PluginAPI) GetUserByUsername(name string) (*model.User, *model.AppErr
 	return api.app.GetUserByUsername(name)
 }
 
+func (api *PluginAPI) GetUsersByUsernames(usernames []string, asAdmin bool) ([]*model.User, *model.AppError) {
```
Comment (jwilander) [inline app/plugin_api.go]: I think it makes sense to remove the parameter for `asAdmin` and just always use true in the app method call. All it's doing is sanitizing the returned profiles which plugins can do themselves if needed and I'd rather keep the API simpler where possible.

@lieut-data @czertbytes thoughts?

Comment (czertbytes) [reply app/plugin_api.go]: I am very new to pluginAPI but generally it makes sense to me to remove `asAdmin` parameter and use to `true` in method call. 👍

Comment (lieut-data) [reply app/plugin_api.go]: 100% agree that we should simplify the API here, and there's no compelling reason to get users as a non-admin at present.

File: plugin/api.go [hunk 2]
```diff
@@ -52,6 +52,9 @@ type API interface {
 	// GetUserByUsername gets a user by their username.
 	GetUserByUsername(name string) (*model.User, *model.AppError)
 
+	// GetUsersByUsernames gets a users by their usernames.
```
Comment (lieut-data) [inline plugin/api.go]: nit: Can you fix the grammar here, and add a minimum server version documenting same? (We literally started that yesterday, so not to worry for missing it!)
```suggestion
	// GetUsersByUsernames gets users by their usernames.
	//
	// Minimum server version: 5.6
```

**Pull request merged**